src/android/FileHelper.java

```diff
@@ -79,50 +79,72 @@ public static String getRealPath(String uriString, CordovaInterface cordova) {
     @SuppressLint("NewApi")
     public static String getRealPathFromURI_API19(Context context, Uri uri) {
         String filePath = "";
+	Cursor cursor = null;
+
         try {
-            String wholeID = DocumentsContract.getDocumentId(uri);
+             String wholeID = DocumentsContract.getDocumentId(uri);
 
             // Split at colon, use second item in the array
             String id = wholeID.indexOf(":") > -1 ? wholeID.split(":")[1] : wholeID.indexOf(";") > -1 ? wholeID
                     .split(";")[1] : wholeID;
 
-            String[] column = { MediaStore.Images.Media.DATA };
+	    // @dlogo 20151210 Get type to check wheter is image or video
+	    String type = wholeID.indexOf(":") > -1 ? wholeID.split(":")[0] : wholeID.indexOf(";") > -1 ? wholeID
+                    .split(";")[0] : wholeID;
+
+	    Uri contentUri = null;
+            if ("image".equals(type)) {
+                contentUri = MediaStore.Images.Media.EXTERNAL_CONTENT_URI;
+            } else if ("video".equals(type)) {
+                contentUri = MediaStore.Video.Media.EXTERNAL_CONTENT_URI;
+            } else if ("audio".equals(type)) {
+                contentUri = MediaStore.Audio.Media.EXTERNAL_CONTENT_URI;
+            }
+
+	    String[] column = { "_data" };
```

Comment:
Instead of hardcoding "_data", it looks like you can reference it from the base class that images/video/audio inherit from http://developer.android.com/reference/android/provider/MediaStore.MediaColumns.html